src/decaf.cpp

```diff
@@ -24,4 +38,43 @@ namespace decaf {
       exec = visit(inst);
     } while (exec == ExecutionResult::Continue);
   }
+
+  ExecutionResult Interpreter::visitAdd(llvm::BinaryOperator& op) {
```

Comment:
Do you think we should copy paste this logic for `sub` or create an abstraction here? I'm good with either but curious what you think

Reply:
For now this code is rather small (5 lines) so I think we should just copy-paste. If the implementations here get larger then it would be worth abstracting it out.